sql/core/src/main/scala/org/apache/spark/sql/execution/python/ExtractPythonUDFs.scala

```diff
@@ -166,3 +174,40 @@ object ExtractPythonUDFs extends Rule[SparkPlan] {
     }
   }
 }
+
+// This rule is to push deterministic predicates through BatchEvalPythonExec
+object PushPredicateThroughBatchEvalPython extends Rule[SparkPlan] with PredicateHelper {
```

Comment:
Most of codes are from the optimizer rule `PushDownPredicate`. Not sure whether we should combine them. You know, this rule is for `SparkPlan`.